Userland/Libraries/LibJS/Runtime/Temporal/AbstractOperations.cpp

```diff
@@ -1318,23 +1344,15 @@ ThrowCompletionOr<TemporalInstant> parse_temporal_instant_string(VM& vm, String
 }
 
 // 13.30 ParseTemporalZonedDateTimeString ( isoString ), https://tc39.es/proposal-temporal/#sec-temporal-parsetemporalzoneddatetimestring
-ThrowCompletionOr<TemporalZonedDateTime> parse_temporal_zoned_date_time_string(VM& vm, String const& iso_string)
+ThrowCompletionOr<ISODateTime> parse_temporal_zoned_date_time_string(VM& vm, String const& iso_string)
 {
     // 1. If ParseText(StringToCodePoints(isoString), TemporalZonedDateTimeString) is a List of errors, throw a RangeError exception.
     auto parse_result = parse_iso8601(Production::TemporalZonedDateTimeString, iso_string);
     if (!parse_result.has_value())
         return vm.throw_completion<RangeError>(ErrorType::TemporalInvalidZonedDateTimeString, iso_string);
 
-    // 2. Let result be ? ParseISODateTime(isoString).
-    auto result = TRY(parse_iso_date_time(vm, *parse_result));
-
-    // 3. Let timeZoneResult be ? ParseTemporalTimeZoneString(isoString).
-    auto time_zone_result = TRY(parse_temporal_time_zone_string(vm, iso_string));
-
-    // 4. Return the Record { [[Year]]: result.[[Year]], [[Month]]: result.[[Month]], [[Day]]: result.[[Day]], [[Hour]]: result.[[Hour]], [[Minute]]: result.[[Minute]], [[Second]]: result.[[Second]], [[Millisecond]]: result.[[Millisecond]], [[Microsecond]]: result.[[Microsecond]], [[Nanosecond]]: result.[[Nanosecond]], [[Calendar]]: result.[[Calendar]], [[TimeZoneZ]]: timeZoneResult.[[Z]], [[TimeZoneOffsetString]]: timeZoneResult.[[OffsetString]], [[TimeZoneName]]: timeZoneResult.[[Name]] }.
-    // NOTE: This returns the two structs together instead of separated to avoid a copy in ToTemporalZonedDateTime, as the spec tries to put the result of InterpretTemporalDateTimeFields and ParseTemporalZonedDateTimeString into the same `result` variable.
-    // InterpretTemporalDateTimeFields returns an ISODateTime, so the moved in `result` here is subsequently moved into ParseTemporalZonedDateTimeString's `result` variable.
-    return TemporalZonedDateTime { .date_time = move(result), .time_zone = move(time_zone_result) };
+    // 2. Return ? ParseISODateTime(isoString).
+    return TRY(parse_iso_date_time(vm, *parse_result));
```

Comment:
You can omit `TRY` at the end of a function given `parse_iso_date_time` returns the same type - saves us one `ThrowCompletionOr` unwrap.

Reply:
Yeah sure